hilti/include/rt/types/vector.h

```diff
@@ -136,14 +136,21 @@ class Vector : public std::vector<T, Allocator>, public hilti::rt::detail::itera
     Vector& operator=(const Vector&) = default;
     Vector& operator=(Vector&&) noexcept = default;
 
-    const T& operator[](uint64_t i) const {
+    const T& operator[](uint64_t i) const& {
         if ( i >= V::size() )
             throw IndexError(fmt("vector index %" PRIu64 " out of range", i));
 
         return V::data()[i];
     }
 
-    auto operator[](uint64_t i);
+    T operator[](uint64_t i) && {
```

Comment:
I'm learning something here! I did not know that one can overload based on the type of reference, nice.

Reply:
Yes, very useful to make proxy classes safer.